packages/trackers/lib/index.ts

```diff
@@ -0,0 +1,113 @@
+import { getNodeTree, matchPathPattern } from "./utils/index.js";
+import { fetchEvents } from "./stores/index.js";
+import { IEvent, TrackerContext } from "./interfaces/index.js";
+
+export class Trackers {
+  observer: MutationObserver | null;
+  track: TrackerContext["track"];
+  events: IEvent[];
+  store: TrackerContext["store"];
+  debug: boolean;
+
+  constructor(context: TrackerContext) {
+    if (!Boolean(context.track)) throw new Error("Missing required `track` method!");
+
+    this.store = context.store;
+    this.observer = null;
+    this.track = context.track;
+    this.events = [];
+    this.debug = Boolean(context.debug);
+  }
+
+  /**
+   * Fetches events and starts registering observers to begin tracking
+   * note: optionally can pass events, if not they'll be fetched from the store config
+   * @param {IEvent[]} events (optional) - array of events to perform the tracking for
+   **/
+  async initialize(events?: IEvent[]) {
+    if ((window as any).fwTrackersRegistered) return;
+
+    if (!events?.length && !this.store) return console.error("Malformed configuration. Please provide store config or list of events");
```

Comment:
Perhaps, we can throw error?

Reply:
wanted to keep this a passive library as this isn't critical so it'll warn instead of throwing errors

Reply:
But the library can still throw it - client can treat it passive - catch and ignore?